core/src/main/java/org/apache/seata/core/rpc/netty/AbstractNettyRemoting.java

```diff
@@ -173,6 +175,12 @@ protected Object sendSync(Channel channel, RpcMessage rpcMessage, long timeoutMi
             LOGGER.warn("sendSync nothing, caused by null channel.");
             return null;
         }
+        if (MsgVersionHelper.versionNotSupport(channel, rpcMessage)) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Message sending will be skipped as the client version does not support it,{}", rpcMessage);
+            }
+            return new VersionNotSupportMessage();
```

Comment:
Why not throw an exception?

Reply:
If an exception is thrown, the server may abort the current execution. In that case, the message-sending step should be skipped, yet still considered successfully completed.